nixos/modules/services/network-filesystems/samba.nix

```diff
@@ -3,232 +3,162 @@
 with lib;
 
 let
-
-  smbToString = x: if builtins.typeOf x == "bool"
-                   then boolToString x
-                   else toString x;
-
   cfg = config.services.samba;
 
-  samba = cfg.package;
-
-  shareConfig = name:
-    let share = getAttr name cfg.shares; in
-    "[${name}]\n " + (smbToString (
-       map
-         (key: "${key} = ${smbToString (getAttr key share)}\n")
-         (attrNames share)
-    ));
-
-  configFile = pkgs.writeText "smb.conf"
-    (if cfg.configText != null then cfg.configText else
-    ''
-      [global]
-      security = ${cfg.securityType}
-      passwd program = /run/wrappers/bin/passwd %u
-      invalid users = ${smbToString cfg.invalidUsers}
-
-      ${cfg.extraConfig}
-
-      ${smbToString (map shareConfig (attrNames cfg.shares))}
-    '');
-
-  # This may include nss_ldap, needed for samba if it has to use ldap.
-  nssModulesPath = config.system.nssModules.path;
-
-  daemonService = appName: args:
-    { description = "Samba Service Daemon ${appName}";
-
-      after = [ (mkIf (cfg.enableNmbd && "${appName}" == "smbd") "samba-nmbd.service") "network.target" ];
-      requiredBy = [ "samba.target" ];
-      partOf = [ "samba.target" ];
-
-      environment = {
-        LD_LIBRARY_PATH = nssModulesPath;
-        LOCALE_ARCHIVE = "/run/current-system/sw/lib/locale/locale-archive";
-      };
-
-      serviceConfig = {
-        ExecStart = "${samba}/sbin/${appName} --foreground --no-process-group ${args}";
-        ExecReload = "${pkgs.coreutils}/bin/kill -HUP $MAINPID";
-        LimitNOFILE = 16384;
-        PIDFile = "/run/${appName}.pid";
-        Type = "notify";
-        NotifyAccess = "all"; #may not do anything...
-        Slice = "system-samba.slice";
-      };
-      unitConfig.RequiresMountsFor = "/var/lib/samba";
-
-      restartTriggers = [ configFile ];
-    };
+  settingsFormat = pkgs.formats.ini { };
```

Comment:
Samba settings aren't strict INI so this breaks things - specifically configuring extensions like `fruit:appl = yes` for enabling the Darwin compatibility layer.

Reply:
Hello! Thank you for the feedback.

Could you provide an example where this configuration isn’t working as expected, please?

To configure `fruit:appl = yes` for `vfs_fruit`, you can use this syntax:

```nix
services.samba.settings.global."fruit:appl" = "yes";
```